### ISSUE TITLE
No ConfigurationProperties annotation found on  'org.springframework.cloud.client.loadbalancer.LoadBalancerProperties' after 2021 upgrade

### ISSUE THREAD

**Issue by sta-szek** [spring-cloud/spring-cloud-kubernetes#931]
**Describe the bug**

The issue details:

```
Caused by: java.lang.IllegalStateException: No ConfigurationProperties annotation found on  'org.springframework.cloud.client.loadbalancer.LoadBalancerProperties'.
	at org.springframework.util.Assert.state(Assert.java:97) ~[spring-core-5.3.13.jar:5.3.13]
	at org.springframework.boot.context.properties.ConfigurationPropertiesBeanRegistrar.registerBeanDefinition(ConfigurationPropertiesBeanRegistrar.java:86) ~[spring-boot-2.6.1.jar:2.6.1]
	at org.springframework.boot.context.properties.ConfigurationPropertiesBeanRegistrar.register(ConfigurationPropertiesBeanRegistrar.java:60) ~[spring-boot-2.6.1.jar:2.6.1]
	at org.springframework.boot.context.properties.ConfigurationPropertiesBeanRegistrar.register(ConfigurationPropertiesBeanRegistrar.java:54) ~[spring-boot-2.6.1.jar:2.6.1]
	at java.base/java.lang.Iterable.forEach(Iterable.java:75) ~[na:na]
	at org.springframework.boot.context.properties.EnableConfigurationPropertiesRegistrar.registerBeanDefinitions(EnableConfigurationPropertiesRegistrar.java:49) ~[spring-boot-2.6.1.jar:2.6.1]
	at org.springframework.context.annotation.ImportBeanDefinitionRegistrar.registerBeanDefinitions(ImportBeanDefinitionRegistrar.java:86) ~[spring-context-5.3.13.jar:5.3.13]
	at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.lambda$loadBeanDefinitionsFromRegistrars$1(ConfigurationClassBeanDefinitionReader.java:396) ~[spring-context-5.3.13.jar:5.3.13]
	at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:684) ~[na:na]
	at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.loadBeanDefinitionsFromRegistrars(ConfigurationClassBeanDefinitionReader.java:395) ~[spring-context-5.3.13.jar:5.3.13]
	at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.loadBeanDefinitionsForConfigurationClass(ConfigurationClassBeanDefinitionReader.java:157) ~[spring-context-5.3.13.jar:5.3.13]
	at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.loadBeanDefinitions(ConfigurationClassBeanDefinitionReader.java:129) ~[spring-context-5.3.13.jar:5.3.13]
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.processConfigBeanDefinitions(ConfigurationClassPostProcessor.java:343) ~[spring-context-5.3.13.jar:5.3.13]
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.postProcessBeanDefinitionRegistry(ConfigurationClassPostProcessor.java:247) ~[spring-context-5.3.13.jar:5.3.13]
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanDefinitionRegistryPostProcessors(PostProcessorRegistrationDelegate.java:311) ~[spring-context-5.3.13.jar:5.3.13]
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:112) ~[spring-context-5.3.13.jar:5.3.13]
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:746) ~[spring-context-5.3.13.jar:5.3.13]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:564) ~[spring-context-5.3.13.jar:5.3.13]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:730) ~[spring-boot-2.6.1.jar:2.6.1]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:412) ~[spring-boot-2.6.1.jar:2.6.1]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:302) ~[spring-boot-2.6.1.jar:2.6.1]
	at org.springframework.boot.test.context.SpringBootContextLoader.loadContext(SpringBootContextLoader.java:121) ~[spring-boot-test-2.6.1.jar:2.6.1]
	at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContextInternal(DefaultCacheAwareContextLoaderDelegate.java:99) ~[spring-test-5.3.13.jar:5.3.13]
	at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContext(DefaultCacheAwareContextLoaderDelegate.java:124) ~[spring-test-5.3.13.jar:5.3.13]
	... 72 common frames omitted
```

pom.xml:
spring-boot 2.6.1
spring-cloud 2021.0.0

```
    <properties>
        <spring-cloud-kubernetes-leader.version>1.1.10.RELEASE</spring-cloud-kubernetes-leader.version>
        <starter-test.version>2.15.0-SNAPSHOT</starter-test.version>
    </properties>
    <dependencies>

        <!-- SPRING-CLOUD -->
        <dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-starter-openfeign</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-starter-loadbalancer</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-kubernetes-leader</artifactId>
            <version>${spring-cloud-kubernetes-leader.version}</version>
        </dependency>
```

**Sample**
Unfortunatelly cannot, we have custom parent shared by starter-parent and all microservices.
starter-parent is used by all other starters

the issue affect our `starter-cloud` (inherits from starter-parent, inherits from cloud-parent):

```
        <spring-boot.version>2.6.1</spring-boot.version>
        <spring-cloud.version>2021.0.0</spring-cloud.version>
        <spring-cloud-security.version>2.2.5.RELEASE</spring-cloud-security.version>
        <spring-security-oauth2.version>2.5.1.RELEASE</spring-security-oauth2.version>
        <spring-security-test.version>5.6.0</spring-security-test.version>
```

When i remove:
```
        <dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-starter-loadbalancer</artifactId>
        </dependency>
```
i see no issue

**Comment by ryanjbaxter** [comment]
A minimal sample that reproduces the issue is all we need, we don't need your exact project

**Comment by sta-szek** [comment]
Can it be the reason?

```
mvn dependency:tree -Dincludes=org.springframework.cloud:spring-cloud-openfeign-core
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ cloud-poc ---
[INFO] com.sap.fsm:cloud-poc:jar:2.4.0
[INFO] \- com.sap.fsm.springboot.starter:starter-cloud:jar:2.15.0-SNAPSHOT:compile
[INFO]    \- org.springframework.cloud:spring-cloud-starter-openfeign:jar:3.1.0:compile
[INFO]       \- org.springframework.cloud:spring-cloud-openfeign-core:jar:3.0.5:compile
```

```
// omit imports
@Configuration(proxyBeanMethods = false)
@EnableConfigurationProperties(LoadBalancerProperties.class) <----- this is invalid. it should be LoadBalancerClientsProperties.class IMO
class DefaultFeignLoadBalancerConfiguration {

```

**Comment by sta-szek** [comment]
yes, confirmed ⬆️ 

so i excluded `spring-cloud-starter-openfeign` and included hardcoded version 3.1.0 and now it works...

**Comment by sta-szek** [comment]
sounds like issue on my side, beacuse in mvn dependenies i see you have proper version:
https://mvnrepository.com/artifact/org.springframework.cloud/spring-cloud-starter-openfeign/3.1.0
so i close the task and start digging into dependency hell on my side 👍
